### PR TITLE
chore: remove ProcessBackendTLSPoliciesAncestorRef

### PR DESCRIPTION
**Pull request by zhaohuabing**
`ProcessBackendTLSPoliciesAncestorRef` checks whether the `Gateway` in `policy.Status.Ancestors` exists, if not, it removes that Ancestor.

This might be problematic if multiple controllers are reconciling the `BackendTLSPolicy`, as the `policy.Status.Ancestors` could contain valid `Gateway` references from another controller.

### CODE REVIEW

**Review by zirain — COMMENTED** [review]
LGTM, defer to @arkodg

**Comment by shawnh2** [comment]
I've been working on adding `PolicyStatus` for xPolicy recently, like in #2846. Maybe `BackendTLSPolicy` can also be handled via this PR.

**Comment by zhaohuabing** [comment]
> I've been working on adding `PolicyStatus` for xPolicy recently, like in #2846. Maybe `BackendTLSPolicy` can also be handled via this PR.

`BackendTLSPolicy` already adapted `gwapiv1a2.PolicyStatus`. This PR is trying to fix another issue.

File: internal/provider/kubernetes/controller.go [hunk 1]
```diff
@@ -873,9 +873,9 @@ func (r *gatewayAPIReconciler) processBackendTLSPolicies(
 
 	for _, policy := range backendTLSPolicies.Items {
 		policy := policy
-		// Discard Status to reduce memory consumption in watchable
-		// It will be recomputed by the gateway-api layer
-		policy.Status = gwapiv1a2.PolicyStatus{} // todo ?
+		// The status of BackendTLSPolicies can't be discarded because the status
```
Comment (arkodg) [inline internal/provider/kubernetes/controller.go:876]: can or can't ?

Comment (zhaohuabing) [reply internal/provider/kubernetes/controller.go:876]: IMO can't

**Review by arkodg — APPROVED** [review]

**Pull request merged**